merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/AutoValueMappers.java

```diff
@@ -0,0 +1,166 @@
+package gov.nasa.jpl.aerie.merlin.processor;
+
+import com.squareup.javapoet.AnnotationSpec;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import com.squareup.javapoet.TypeVariableName;
+import gov.nasa.jpl.aerie.contrib.serialization.mappers.RecordValueMapper;
+import gov.nasa.jpl.aerie.merlin.framework.ValueMapper;
+import gov.nasa.jpl.aerie.merlin.framework.annotations.AutoValueMapper;
+import gov.nasa.jpl.aerie.merlin.processor.metamodel.MissionModelRecord;
+import gov.nasa.jpl.aerie.merlin.processor.metamodel.TypeRule;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Parameterizable;
+import javax.lang.model.element.RecordComponentElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeMirror;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class AutoValueMappers {
+  static TypeRule typeRule(final Element autoValueMapperElement, final ClassName generatedClassName) throws InvalidMissionModelException {
+    if (!autoValueMapperElement.getKind().equals(ElementKind.RECORD)) {
+      throw new InvalidMissionModelException(
+          "@%s.%s is only allowed on records".formatted(
+              AutoValueMapper.class.getSimpleName(),
+              AutoValueMapper.Record.class.getSimpleName()),
+          autoValueMapperElement);
```

Comment:
You had to hard-code the names of `AutoValueMapper` and `Record` anyway, might as well inline it into the string.

Reply:
If we do an IDE-based refactor of `AutoValueMapper` to some other name, ideally we want the IDE to find all use-sites. There's no real guarantee it'll find references that are just sequences of characters in string literals, since those aren't true references at the language level.

If there are other places where these names are hardcoded into strings, we should probably fix those.

Reply:
Yes - here is the comment from the original PR where Jonathan asked me to keep the classname where the IDE can see it:

https://github.com/NASA-AMMOS/aerie/pull/188#discussion_r889611173